crates/web-plugins/didcomm-messaging/src/web/dispatcher.rs

```diff
@@ -23,67 +21,85 @@ pub(crate) async fn process_didcomm_message(
     State(state): State<Arc<AppState>>,
     Extension(message): Extension<Message>,
 ) -> Response {
-    if message.type_ == MEDIATE_FORWARD_2_0 {
-        return match mediator_forward_process(&state, message).await {
-            Ok(_message) => StatusCode::ACCEPTED.into_response(),
-            Err(response) => response,
-        };
-    }
-    let response = match message.type_.as_str() {
-        KEYLIST_UPDATE_2_0 => {
-            web::handler::stateful::process_plain_keylist_update_message(
-                Arc::clone(&state),
-                message,
-            )
-            .await
-        }
-        KEYLIST_QUERY_2_0 => {
-            web::handler::stateful::process_plain_keylist_query_message(Arc::clone(&state), message)
+    let response: Result<Option<Message>, Response> = match message.type_.as_str() {
+        MEDIATE_FORWARD_2_0 => {
+            forward::web::handler::mediator_forward_process(state.clone(), message)
```

Comment:
It is important that handlers really have the **same** signature. There should a base error that can uniformly be turned into an axum response. That said, you should be able to have `.map_err(|e| e.into_response())` once.
```rust
let response: Result<Option<Message>, Error> = match message.type_.as_str() {
```

Reply:
The choice of decoupling the errors has been motivated by many decisions:
* The protocols are independent and designed as individual crates. Each crate becomes self-contained, and changes to one protocol’s error handling don’t affect others. Other crates wouldn't be forced to import errors that they don't need or are not aware of.
* Decoupling the errors enable us to handle errors that are specific to the corresponding protocol only. Thus, we can remove, add or modify existing protocol at any time without touching anything else.
* It would be easy to maintain since in case of error, we can immediately recognize the scope of each error type.
* With unified error type, any change would require a recompilation of each crate that uses it.

As we design a collector to gather all loaded protocols and their handlers, the need for repeatedly calling `into_response()` for each error will be eliminated. We will obtain the appropriate handler from the collector based on the message type, allowing us to call `into_response()` just once for the corresponding handler.

Reply:
> I'm not asking to use the same error, but the same base error. May I know what final signature did you agree with @ndefokou for the handlers? Consider using it here.

You're referring to a general `Error` type that encompasses the errors from all protocols, right? If so, that's precisely what I'm addressing here.

Reply:
Yes, exactly. Do you mean you're providing it in this PR? What is its name pls? I could not identify it. It should enforce the IntoResponse trait.

Reply:
> Yes, exactly. Do you mean you're providing it in this PR? What is its name pls? I could not identify it. It should enforce the IntoResponse trait.

No I meant that’s what my previous response was trying to explain. Why I’ve chosen this error handling design instead of the shared error type.

Reply:
We're still not getting each other. Let's resume the discussion on #219 when the feature branch will be updated with these changes.